source/fem-dem/cfd_dem_coupling.cc

```diff
@@ -413,15 +701,87 @@ CFDDEMSolver<dim>::check_contact_search_step_dynamic(const unsigned int &)
   return contact_detection_step;
 }
 
+template <int dim>
+inline bool
+CFDDEMSolver<dim>::no_load_balance(const unsigned int &)
+{
+  return false;
+}
+
+template <int dim>
+inline bool
+CFDDEMSolver<dim>::check_load_balance_once(const unsigned int &)
+{
+  bool load_balance_step = (this->simulation_control->get_step_number() ==
+                            this->cfd_dem_simulation_parameters.dem_parameters
+                              .model_parameters.load_balance_step);
+
+  if (load_balance_step || checkpoint_step)
+    {
+      load_balance();
+    }
```

Comment:
See next comment. Move call to load_balance outside and call it using the value of the boolean.
You could also then move this function out of the class and keep a single function for CFD-DEM and DEM and refactor it.

Reply:
done